src/app/ReadClient.h

```diff
@@ -365,7 +377,16 @@ class ReadClient : public Messaging::ExchangeDelegate
     CHIP_ERROR ProcessAttributePath(AttributePathIB::Parser & aAttributePath, ConcreteDataAttributePath & aClusterInfo);
     CHIP_ERROR ProcessReportData(System::PacketBufferHandle && aPayload);
     const char * GetStateStr() const;
-    bool ResubscribeIfNeeded();
+
+    /*
+     * Called internally to check if the client needs to resubscribe based upon the resubscribe policy, schedule when to re-subscribe
+     * next time, and update aResubscribe and aNextResubscribeIntervalMsec accordingly.
```

Comment:
_"Checks if we should re-subscribe based on the specified re-subscription policy. If we should, re-subscription is scheduled and aResubscribe and aNextResubscribeIntervalMsec are updated accordingly."_